src/include/schema.h

```diff
@@ -113,12 +112,16 @@ struct __wt_table {
 } while (0)
 
 /*
- * WT_WITH_CHECKPOINT_LOCK --
+ * WT_WITH_CHECKPOINT_LOCK, WT_WITH_CHECKPOINT_LOCK_NOWAIT --
  *	Acquire the checkpoint lock, perform an operation, drop the lock.
  */
-#define	WT_WITH_CHECKPOINT_LOCK(session, ret, op)			\
-	WT_WITH_LOCK(session, ret,					\
+#define	__WT_WITH_CHECKPOINT_LOCK(session, ret, nowait, op)		\
```

Comment:
Can we find some way to do this without adding a new class of macro that starts with underscores?